libdnf/repo/Repo-private.hpp

```diff
@@ -176,6 +177,7 @@ class Repo::Impl {
     int main_nsolvables{0};
     int main_nrepodata{0};
     int main_end{0};
+    std::mutex attachLibsolvMutex;
```

Comment:
Add a comment explaining what it does, for example:
Lock attachLibsolvRepo() and detachLibsolvRepo() to ensure atomic behavior in threaded environment such as PackageKit.